src/Diagnostics.RuntimeHost/Utilities/CredentialTapper.cs

```diff
@@ -0,0 +1,44 @@
+﻿using System;
+
+namespace Diagnostics.RuntimeHost.Utilities
+{
+    /// <summary>
+    /// Traps potential creds and obfuscate them.
+    /// </summary>
+    public static class CredentialTrapper
```

Comment:
>CredentialTrapper [](start = 24, length = 17)

This is a bit unrelated to this PR but we have similar code in GeoMasterDataProvider.cs. Can you update that class also to use CredentialTapper?

Reply:
I found the code in Geomaster Data Provider. You created this method as Public:
public string RemovePIIFromSettings(string content) {... }

Any reason for that? I can remove it but I am not sure if the intention was to use it in detectors as well (in addition to the gist) since the method is Public.

Reply:
Feel free to remove. I think the detectors are using the gist

Reply:
Updated the PR